physical/raft/fsm.go

```diff
@@ -404,57 +384,99 @@ func (f *FSM) Transaction(ctx context.Context, txns []*physical.TxnEntry) error
 	return err
 }
 
-// Apply will apply a log value to the FSM. This is called from the raft
+// ApplyBatch will apply a set of logs to the FSM. This is called from the raft
 // library.
-func (f *FSM) Apply(log *raft.Log) interface{} {
-	command := &LogData{}
-	err := proto.Unmarshal(log.Data, command)
-	if err != nil {
-		f.logger.Error("error proto unmarshaling log data", "error", err)
-		panic("error proto unmarshaling log data")
-	}
+func (f *FSM) ApplyBatch(logs []*raft.Log) []interface{} {
+	if len(logs) == 0 {
+		return []interface{}{}
+	}
+
+	// Do the unmarshalling first so we don't hold locks
+	var latestConfiguration *ConfigurationValue
+	commands := make([]interface{}, 0, len(logs))
+	for _, log := range logs {
+		switch log.Type {
+		case raft.LogCommand:
+			command := &LogData{}
+			err := proto.Unmarshal(log.Data, command)
+			if err != nil {
+				f.logger.Error("error proto unmarshaling log data", "error", err)
+				panic(len(log.Data))
```

Comment:
Is the `panic` here intended? Will the panic below executed at all?

Reply:
Sure isn't, left in from some debugging